src/pages/Editor.vue

```diff
@@ -839,25 +850,28 @@ export default {
       }
 
       // create item, then create the question, then update local states
-      ItemService.createItem({
+      ItemAPIService.createItem({
         plio: this.plioDBId,
         type: this.getItemTypeForNewItem(),
         time: newTimestamp,
         meta: this.getMetadataForNewItem(),
       })
         .then((createdItem) => {
+          // storing the newly created item into "newItem"
           newItem = createdItem;
 
           if (createdItem.type == "question") {
-            return QuestionService.createQuestion({
+            return QuestionAPIService.createQuestion({
               item: createdItem.id,
-              type: "mcq",
-              options: ["", ""],
-              correct_answer: "0",
+              text: this.getDetailsForNewQuestion().text,
+              type: this.getQuestionTypeForNewQuestion(),
+              options: this.getDetailsForNewQuestion().options,
+              correct_answer: this.getDetailsForNewQuestion().correct_answer,
```

Comment:
why call the function 4 times -> the keys are the same so just pass the value of that function to the API no?